build/gen_stub.php

```diff
@@ -629,26 +662,332 @@ private function getFlagsAsString(): string
+    public $type;
+    /** @var Expr|null */
+    public $defaultValue;
+
+    public function __construct(PropertyName $name, int $flags, bool $isKnownName, bool $isDeprecated, ?Type $type, ?Expr $defaultValue)
+    {
+        $this->name = $name;
+        $this->flags = $flags;
+        $this->isKnownName = $isKnownName;
+        $this->isDeprecated = $isDeprecated;
+        $this->type = $type;
+        $this->defaultValue = $defaultValue;
+    }
 
     public function discardInfoForOldPhpVersions(): void {
-        $this->return->type = null;
-        foreach ($this->args as $arg) {
-            $arg->type = null;
-            $arg->defaultValue = null;
+        $this->type = null;
+    }
+
+    public function getDeclaration(): string {
+        $code = "\\\n";
+
+        $propertyName = $this->name->property;
+
+        $defaultValueConstant = false;
+        if ($this->defaultValue === null) {
+            $defaultValue = null;
+            $defaultValueType = "undefined";
+        } else {
+            $evaluator = new ConstExprEvaluator(
+                function (Expr $expr) use (&$defaultValueConstant) {
+                    if ($expr instanceof Expr\ConstFetch) {
+                        $defaultValueConstant = true;
+                        return null;
+                    }
+
+                    throw new Exception("Property $this->name has an unsupported default value");
+                }
+            );
+            $defaultValue = $evaluator->evaluateDirectly($this->defaultValue);
+            $defaultValueType = gettype($defaultValue);
+        }
+
+        if ($defaultValueConstant) {
+            echo "Skipping code generation for property $this->name, because it has a constant default value\n";
+            return "";
         }
+
+        if ($this->type) {
+            $typeFlags = $this->type->tryToRepresentableType();
+            if ($typeFlags === null) {
+                throw new Exception("Unimplemented property type");
+            }
+
+            if ($typeFlags->classType) {
+                $simpleType = $this->type->tryToSimpleType();
+                if ($simpleType) {
+                    $typeCode = "(zend_type) ZEND_TYPE_INIT_CE(class_entry_" . $typeFlags->classType->toEscapedName() . ", " .  ((int) $this->type->isNullable()) . ", 0)";
+                } else {
+                    throw new Exception("Property $this->name has an unsupported union type");
+                }
+            } else {
+                $typeCode = "(zend_type) ZEND_TYPE_INIT_MASK(" . $typeFlags->toTypeMask() . ")";
+            }
+
+            $code .= $this->initializeValue($defaultValueType, $defaultValue);
+
+            if ($this->isKnownName) {
+                $nameCode = "ZSTR_KNOWN(ZEND_STR_" . strtoupper($propertyName) . ")";
+            } else {
+                $code .= "\tzend_string *property_{$propertyName}_name = zend_string_init(\"$propertyName\", sizeof(\"$propertyName\") - 1, 1); \\\n";
```

Comment:
It seemed easiest to just generate a new name and default value variable for each properties... However, I can minimize the number of variables if it's worth the effort (needs some "lookahead" checks).